Update wasm to use the same method as mobile for feature flags

Hinton opened the pull request:
## 🎟️ Tracking

<!-- Paste the link to the Jira or GitHub issue or otherwise describe / point to where this change is coming from. -->

## 📔 Objective

<!-- Describe what the purpose of this PR is, for example what bug you're fixing or new feature you're adding. -->

Updates wasm to use the same implementation as mobile, i.e. `load_flags`. This uses a hashmap instead of the internal `Flags` enum as we want to prevent additions or removal of flags to be a breaking change.

Consumers should pass in the full flag object they get from the config endpoint and let the SDK handle the rest.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation
  team

## 🦮 Reviewer guidelines

<!-- Suggested interactions but feel free to use (or not) as you desire! -->

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

Hinton committed:
Revert "[PM-24243] Add load_flags method to wasm PlatformClient (#369)"

This reverts commit 481349272195b330c52610f64b4b7f3559b0e25b.

Hinton committed:
Expose the already existing load_flags

Hinton committed:
Re-add docs

Hinton committed:
More docs

shane-melton commented:
One thing to note is cipher key encryption feature flag from the config endpoint is of the form `cipher-key-encryption` while the SDK uses `enableCipherKeyEncryption`. We'll need to transform that somewhere, at least for that flag. Android currently does something similar [here.](https://github.com/bitwarden/android/blob/9bfbe0c08724b08f6225777df5d7ace23a5d4ac3/app/src/main/kotlin/com/x8bit/bitwarden/data/platform/manager/FeatureFlagManagerImpl.kt#L21-L24)

Hinton committed:
Add alias to ensure compliance with flags from the config endpoint

Hinton commented:
@shane-melton that should be addressed now. I added a serde alias for the existing value and added default handling to use `kebab-case`.

With this change we should be able to pass in the flags directly from the config endpoint.

dani-garcia commented on crates/bitwarden-wasm-internal/src/platform/mod.rs:
```diff
@@ -21,8 +32,9 @@ impl PlatformClient {
     /// Load feature flags into the client
-    pub fn load_flags(&self, flags: Flags) -> Result<(), JsValue> {
-        self.0.internal.set_flags(&flags);
+    #[wasm_bindgen(js_name = load_flags)]
```
Do we need to set the name? this one should be the default right?

Hinton committed:
Remove wasm rename
